### ISSUE TITLE
Use internal devfile registry as default for parent devfiles

### ISSUE THREAD

**Issue by nils-mosbach** [eclipse/che#21544]
### Is your enhancement related to a problem? Please describe

As described by the devfile spec a devfile can inherit from a parent devfile. That makes things a lot easier, thanks a lot. :)

https://devfile.io/docs/devfile/2.2.0/user-guide/referring-to-a-parent-devfile-in-a-devfile/ e.g. referring to a nodejs environment.

```
schemaVersion: 2.2.0
metadata:
  name: my-project-dev
parent:
    id: redhat/nodejs
    registryUrl: https://devfile-registry.io/
```

We're sharing projects with people that are not within our organization (airgapped) . So providing a devfile registry that's deployed by che that also uses different (local) docker images would be nice. This will also reduce complexity, since developers don't have to type in the base adress of our che instance. If running stage and prod instances of che, its also nice since projects only need to refer to an id  (e.g. `node`) instead of a url.

### Describe the solution you'd like

`parent.registryUrl` could be defaulted to the one deployed by che (e.g. `https://che-test.company.dev/devfile-registry/`).

so basically this Devfile will work.

```
schemaVersion: 2.2.0
metadata:
  name: my-project-dev
parent:
    id: redhat/nodejs
```
